mirroroperator/registrymirror.py

```diff
@@ -254,64 +307,55 @@ def generate_daemon_set(self, daemon_set):
         return service_headless
 
-    def handle_proxy_credentials(self, env):
+    def get_upstream_credentials(self):
         credentials_secret = None
         if self.credentials_secret_name:
             credentials_secret = self.run_action_and_parse_error(self.core_api.read_namespaced_secret,
-                                                             self.credentials_secret_name,
-                                                             self.namespace)
+                                                                 self.credentials_secret_name,
+                                                                 self.namespace)
         if not credentials_secret:
             LOGGER.error("No secret named %s was found in the %s namespace, will use unauth access",
                          self.credentials_secret_name, self.namespace)
-            return env
+            return None
 
         encoded_user = credentials_secret.data.get("username")
         encoded_pass = credentials_secret.data.get("password")
 
         if not (encoded_user and encoded_pass):
             LOGGER.error("Secret %s does not contain username/password",
                          self.credentials_secret_name)
-            return env
-        env.append(client.V1EnvVar(name="REGISTRY_PROXY_USERNAME",
-                                   value=None,
-                                   value_from=client.V1EnvVarSource(
-                                        secret_key_ref=client.V1SecretKeySelector(
-                                            key="username",
-                                            name=self.credentials_secret_name
-                                        )
-                                   ))
-                  )
-        env.append(client.V1EnvVar(name="REGISTRY_PROXY_PASSWORD",
-                                   value=None,
-                                   value_from=client.V1EnvVarSource(
-                                        secret_key_ref=client.V1SecretKeySelector(
-                                            key="password",
-                                            name=self.credentials_secret_name
-                                        )
-                                   ))
-                  )
-        LOGGER.info("Secret selected + env vars set successfully")
-        return env
-
+            return None
+        the_user = base64.b64decode(encoded_user.encode()).decode()
+        the_pass = base64.b64decode(encoded_pass.encode()).decode()
+        LOGGER.info("Secret selected successfully")
+        return (the_user, the_pass)
 
     def generate_stateful_set(self):
-
+        script = '''
+        while true; do
+          TEMPFILE=$(mktemp)
+          cat /etc/ssl/certs/{cert_file} >> $TEMPFILE
+          if [ -d {upstream_cert_dir} ]; then
+            cat {upstream_cert_dir}/{cert_file} >> $TEMPFILE
+          fi
+          mv $TEMPFILE {shared_cert_mount_path}/{cert_file}
+          sleep 1
```

Comment:
Hmmm `sleep 1`?

Either the loop should end once completed, or we should sleep for longer than 1 second.

Seems pretty wasteful with IO otherwise

Reply:
why the loop? Are we expecting the upstream certificate file to change?

Reply:
OK, we are not expecting the file to change. Making this an init container means that we don't need a loop here anyway.